backend/banners/banners.go

```diff
@@ -111,13 +113,29 @@ func (banners *Banners) Init(httpClient *http.Client) {
 
 // Activate invokes showing the message for the given key.
 func (banners *Banners) Activate(key MessageKey) {
+	defer banners.activeLock.Lock()()
 	banners.active[key] = struct{}{}
 	banners.Notify(observable.Event{
 		Subject: "banners/" + string(key),
 		Action:  action.Reload,
 	})
 }
 
+// Deactivate removes the message key from the active map and makes the frontend reload the banner.
+func (banners *Banners) Deactivate(key MessageKey) {
+	defer banners.activeLock.Lock()()
+	_, keyExists := banners.active[key]
+	if !keyExists {
+		banners.log.Errorf("Trying to deactivate unactivated key: %s", key)
+		return
+	}
+	delete(banners.active, key)
+	banners.Notify(observable.Event{
+		Subject: "banners/" + string(key),
+		Action:  action.Reload,
+	})
+}
+
 // GetMessage gets a message for a key if it was activated. nil otherwise, or if no msg exists.
 func (banners *Banners) GetMessage(key MessageKey) *Message {
 	_, active := banners.active[key]
```

Comment:
add `defer banners.activeLock.RLock()()` here as well

Reply:
Sorry, missed that :man_facepalming: Done!